.github/workflows/bump-release-version.yaml

```diff
@@ -0,0 +1,89 @@
+name: Bump Release Version
+
+on:
+  workflow_dispatch:
+    inputs:
+      branch:
+        description: "Release branch to run the workflow against"
+        required: true
+        default: "aptos-release-v1."
+
+permissions:
+  contents: write
+  pull-requests: write
+
+jobs:
+  update-version:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.inputs.branch }}
+
+      - name: Determine next version and update aptos-node/Cargo.toml
+        uses: actions/github-script@v6
+        id: determine-version
+        with:
+          script: |
+            const branch = "${{ github.event.inputs.branch }}".trim();
+
+            // Match everything after 'v' in the branch name
+            const match = branch.match(/aptos-release-v(.*)/);
+            if (!match) {
+              throw new Error(`Branch name does not match expected pattern 'aptos-release-v*'. Got: ${branch}`);
+            }
+            const versionSuffix = match[1]; // Extract everything after 'v'
+            console.log(`Version suffix extracted from branch: ${versionSuffix}`);
+
+            // Read Cargo.toml
+            const fs = require("fs");
+            const cargoFilePath = "aptos-node/Cargo.toml";
+            const cargoContent = fs.readFileSync(cargoFilePath, "utf8");
+
+            // Extract and update the version
+            const versionMatch = cargoContent.match(/version\s*=\s*"(.*?)"/);
+            if (!versionMatch) {
+              throw new Error("Could not find version in Cargo.toml");
+            }
+            const currentVersion = versionMatch[1];
+            console.log(`Current version found in Cargo.toml: ${currentVersion}`);
+
+            let newVersion;
+            if (currentVersion === "0.0.0-main") {
+              newVersion = `${versionSuffix}.0`; // Use the branch suffix for the new version
+            } else {
+              const [major, minor, patch] = currentVersion.split(".").map(Number);
+              newVersion = `${major}.${minor}.${patch + 1}`;
```

Comment:
shall we accept a suffix from the workflow? 
x.y.z can become x.y.z-hotfix or x.y.z+1-rc
x.y.z-rc can become x.y.z
etc

or maybe accept full override?

Reply:
Good question. I wanted to take one step at a time and ignore suffixes this time. A full override might make sense just in case, I will update.